scripts/workspace-conversion/workspace-to-dataset.py

```diff
@@ -0,0 +1,248 @@
+# ------------------------------------------------------------------------------
+# Workspace to Dataset
+#
+# This script converts terra workspaces attributes into a potential catalog entry.
+#
+# Before Running:
+#   python3 -m venv .env
+#   source .env/bin/activate
+#   pip install --upgrade pip
+#   python3 -m pip install requests
+#
+#   Authorization: Access the password in the vault using the appropriate command:
+#      (you will need this for the auth dialog that the script will pull up)
+#   vault login -method=github token=$(cat ~/.github-token)
+#   vault read secret/dsde/terra/kernel/dev/dev/catalog/tests/userAdmin
+#
+# Run:
+#   `python3 workspace-to-dataset.py`
+#   `export GCLOUD_USER={your-email}; export WORKSPACE_NAMESPACE={workspace-namespace}; export WORKSPACE_NAME={workspace-name}; python3 workspace-to-dataset.py`
```

Comment:
Potential future improvements: having command line arguments and validating them?

Reply:
ah definitely would be nice to add